src/Components/Endpoints/src/Builder/RazorComponentEndpointFactory.cs

```diff
@@ -48,6 +49,7 @@ internal class RazorComponentEndpointFactory
         builder.Metadata.Add(HttpMethodsMetadata);
         builder.Metadata.Add(new ComponentTypeMetadata(pageDefinition.Type));
         builder.Metadata.Add(new RootComponentMetadata(rootComponent));
+        builder.Metadata.Add(configuredRenderModesMetadata);
```

Comment:
It looks like that would be less easy since we'd have to flow the list of configured render modes deeper through `PageComponentBuilder` and into the `PageComponentInfo` constructor.

It's internal and doesn't impact any public APIs in any case.